llama-index-packs/llama-index-packs-cohere-citation-chat/llama_index/packs/cohere_citation_chat/base.py

```diff
@@ -0,0 +1,52 @@
+import os
+from typing import Any, Dict, List
+from llama_index.core.llama_pack.base import BaseLlamaPack
+from llama_index.core.chat_engine.types import BaseChatEngine
+from llama_index.core.schema import Document
+from llama_index.core import Settings
+
+from .citations_context_chat_engine import VectorStoreIndexWithCitationsChat
+
+
+class CohereCitationChatEnginePack(BaseLlamaPack):
+    def __init__(self, documents: List[Document], cohere_api_key: str = None) -> None:
+        """Init params."""
+        try:
+            from llama_index.llms.cohere import Cohere
+            from llama_index.embeddings.cohere import CohereEmbedding
+        except ImportError:
+            raise ImportError(
+                "Please run `pip install llama-index-llms-cohere llama-index-embeddings-cohere` "
+                "to use the Cohere."
+            )
+        self.api_key = cohere_api_key or os.environ.get("COHERE_API_KEY")
+        self.llm = Cohere(
+            "command",
+            api_key=self.api_key,
+            temperature=0.5,
+            additional_kwargs={"prompt_truncation": "AUTO"},
+        )
+
+        self.embed_model_query = CohereEmbedding(
+            cohere_api_key=self.api_key,
+            model_name="embed-english-v3.0",
+            input_type="search_query",
+        )
+
+        Settings.llm = self.llm
+        Settings.embed_model = self.embed_model_query
```

Comment:
Modifying global settings under the hood like this will probably be a "surprise" to some users -- probably, we can just pass in the llm/embedding model to the interfaces where they are needed

Reply:
Thank you, it's been resolved.